src/ejabberd_s2s_out.erl

```diff
@@ -285,30 +288,7 @@ open_socket(timeout, StateData) ->
 open_socket(_, StateData) ->
     {next_state, open_socket, StateData}.
 
-%%----------------------------------------------------------------------
-%% IPv4
--spec open_socket1(Host :: binary() | inet:ip_address(),
-                   Port :: inet:port_number()) -> {'error', _} | {'ok', _}.
-open_socket1({_, _, _, _} = Addr, Port) ->
-    open_socket2(inet, Addr, Port);
-%% IPv6
-open_socket1({_, _, _, _, _, _, _, _} = Addr, Port) ->
-    open_socket2(inet6, Addr, Port);
-%% Hostname
-open_socket1(Host, Port) ->
-    lists:foldl(fun(_Family, {ok, _Socket} = R) ->
-                        R;
-                   (Family, _) ->
-                        Addrs = get_addrs(Host, Family),
-                        lists:foldl(fun(_Addr, {ok, _Socket} = R) ->
-                                            R;
-                                       (Addr, _) ->
-                                            open_socket1(Addr, Port)
-                                    end, ?SOCKET_DEFAULT_RESULT, Addrs)
-                end, ?SOCKET_DEFAULT_RESULT, outgoing_s2s_families()).
-
-
--spec open_socket2(Type :: 'inet' | 'inet6',
+-spec open_socket2(Type :: inet | inet6,
```

Comment:
Minor: without `open_socket1` there is only `open_socket2` 😄

Reply:
I just didn't think there was any rule here, because `fun, fun2, fun3`... looks to me just as good as `fun, fun1, fun2`. That's why I decided to minimize the change set.